File: 409_longest_palindrome/409_longest_palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        setOfCh = set()
        cnt = 0
        
        for ch in s:
            if ch not in setOfCh:
                setOfCh.add(ch)
            else:
                setOfCh.remove(ch)
                cnt+=1
        if len(setOfCh) > 0:
            return cnt * 2 + 1
        else:
            return cnt * 2

File: 409_longest_palindrome/409_longest_palindrome.java
import java.util.HashSet;

class Solution {
    public int longestPalindrome(String s) {
        HashSet<Character> setOfCh = new HashSet<Character>();
        int cnt = 0;

        for (char ch : s.toCharArray()) {
            if (!setOfCh.contains(ch)) {
                setOfCh.add(ch);
            } else {
                setOfCh.remove(ch);
                cnt++;
            }
        }
        if (!setOfCh.isEmpty()) {
            return cnt * 2 + 1;
        }
        return cnt * 2;
    }
}